api/api.go

```diff
@@ -114,18 +116,24 @@ func (api *ManagementAPI) TakeSnapshot(w http.ResponseWriter, r *http.Request) {
 
 func getCptvNames(dir string) []string {
 	matches, _ := filepath.Glob(filepath.Join(dir, cptvGlob))
+	failedUploadMatches, _ := filepath.Glob(filepath.Join(dir, "failed-uploads", cptvGlob))
+	matches = append(matches, failedUploadMatches...)
 	names := make([]string, len(matches))
 	for i, filename := range matches {
 		names[i] = filepath.Base(filename)
 	}
 	return names
 }
 
-func checkIfCptvFile(cptv, dir string) bool {
-	for _, n := range getCptvNames(dir) {
-		if n == cptv {
-			return true
+func getRecordingPath(cptv, dir string) string {
```

Comment:
To guard against dodgy cptv file names (e.g. "../../../root/secret.file"), I suggest the following:
* return "" if the cptv file doesn't end in `.cptv`
* Run `filepath.Abs` on each joined path and return "" if a joined path doesn't begin with the value of `dir`.

This will ensure it's not possible to delete or retrieve files outside of the cptv directory :)